x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/evaluation/index.ts

```diff
@@ -66,47 +69,57 @@ export const evaluateAttackDiscovery = async ({
         projectName: string | undefined;
         tracers: LangChainTracer[];
       };
-    }> = connectors.map((connector) => {
-      const llmType = getLlmType(connector.actionTypeId);
+    }> = await Promise.all(
```

Comment:
Some evaluations may take longer to run than the `POST` request handler timeout(s) that start them. Consider removing the newly-added `await Promise.all`, because the caller of this function [notes](https://github.com/elastic/kibana/blob/a108c632a459240f9c27262395e2584f92d43779/x-pack/solutions/security/plugins/elastic_assistant/server/routes/evaluate/post_evaluate.ts#L189):

```typescript
            try {
              // NOTE: we don't wait for the evaluation to finish here, because
              // the client will retry / timeout when evaluations take too long
              void evaluateAttackDiscovery({
                actionsClient,
```

Reply:
hmm... do you have suggestions for how to handle since i need to make an async call for prompts for each connector? should I do that in `post_evaluate.ts` and pass the prompts in keyed by connector id? or is there a way you see to do this remaining in this file?

Reply:
>  since i need to make an async call for prompts for each connector? should I do that in post_evaluate.ts and pass the prompts in keyed by connector id?

sounds good, thank you!

Reply:
fixed!